include/mbedtls/config.h

```diff
@@ -2567,6 +2578,32 @@
  */
 #define MBEDTLS_XTEA_C
 
+/**
+ * \def MBEDTLS_SERIALIZE_C
+ *
+ * Enable the serialization module. This is needed for deligating file system
+ * operations to another process or machine. It helps in on-target testing.
+ * (see MBEDTLS_FS_IO_ALT).
+ *
+ * Module:  library/serialize.c
+ * Caller:  library/net_sockets/serialize.c
+ */
+//#define MBEDTLS_SERIALIZE_C
+
+/**
+ * \def MBEDTLS_SERIALIZE_FORK_FRONTEND_C
+ *
```

Comment:
The frontend code in `serialize.c` calls many standard library functions. It would be good to document what is needed, and to at least state and check the dependency on `! MBEDTLS_PLATFORM_NO_STD_FUNCTIONS`.

Taking a step back, why don't we outsource the frontend code from `serialize.c` to `programs/host/frontend.c`?

Reply:
All this code is guarded under ```MBEDTLS_SERIALIZE_FORK_FRONTEND_C```. It is only used for forking ```frontend.c```. It is not for use within library. Also it is very experimental. Perhaps dependency can be set on ```__linux__```.

Reply:
@gilles-peskine-arm What do you think on that?

Reply:
The frontend is only expected to run on “non-embedded” platforms, so some of the portability considerations for the library proper don't apply to it. In practice, we only care about the frontend on Linux and Windows. In particular, we don't want platform settings from the library configuration to affect the frontend — they don't run on the same platform! Note that for this reason, `config.h` is not used when compiling the frontend.

I'd originally intended fork mode for quick testing of the offloading code. In the real world, I don't expect the target code to launch the frontend: it should either be the other way round (the frontend program arranging to upload the on-target code to the target if necessary) or else there would be common launcher.

I don't think we should document `MBEDTLS_SERIALIZE_FORK_FRONTEND_C` in `config.h` because it's likely to evolve as on-target testing matures. As long as it isn't documented (or it's documented as undocumented), we can stick to whatever works for now, and revisit it later.

@mazimkhan Do you ever use fork mode?